joomla/joomla-cms
# [4.0] Clean up core editors

Pull Request for Issue # .

### Summary of Changes

Remove the deprecated methods from Editors.
Remove the methods from src/Joomla/Editor (needs a deprecation in 3.9)

### WHY?
- The code is largely deprecated since 3.7 (https://github.com/joomla/joomla-cms/pull/12561)
- It's totally wrong to create executable javascript in PHP, PHP should just hand in some parameters and Javascript should execute according to these. Building executable code in PHP is **wrong**

### Testing Instructions
Apply patch, edit an article, play with the xtd-buttons, add content, save. Everything should work.
Repeat the test for Editor none, and Codemirror
Everything should work for these 2 editors apart the closing modal which is a different bug in J4, PR is coming for that later on

### Expected result
No inline, PHP created javascript!

### Actual result
No inline, PHP created javascript but with the same behaviour and functionality as before

### Documentation Changes Required
YES! We need to communicate this shortly with a post


Calling @wilsonge and @mbabker for a decision here
